Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -0,0 +1,255 @@
+
+    if (!validateCSS(cssInfo, outExceptionString))
+        return;
+
+    // FIXME: <https://webkit.org/b/259954> May need to remove unsupported keys.
+    // FIXME: Handle css removal in the UI Process.
+}
+
+void WebExtensionAPIScripting::registerContentScripts(NSObject *scripts, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/registerContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+
+void WebExtensionAPIScripting::getRegisteredContentScripts(NSDictionary *filter, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/getRegisteredContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+void WebExtensionAPIScripting::updateContentScripts(NSObject *scripts, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/updateContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+void WebExtensionAPIScripting::unregisterContentScripts(NSDictionary *filter, Ref<WebExtensionCallbackHandler>&& callback, NSString **outExceptionString)
+{
+    // Documentation: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/unregisterContentScripts
+
+    // FIXME: <https://webkit.org/b/261769> Implement this.
+}
+
+bool WebExtensionAPIScripting::validateScript(NSDictionary *script, NSString **outExceptionString)
+{
+    static NSArray<NSString *> *requiredKeys = @[
+        targetKey,
+    ];
+
+    static NSDictionary<NSString *, id> *keyTypes = @{
+        argsKey: @[ NSObject.class ],
+        argumentsKey: @[ NSObject.class ],
+        filesKey: @[ NSString.class ],
+        funcKey: NSObject.class,
+        functionKey : NSObject.class,
+        targetKey: NSDictionary.class,
+        worldKey: NSString.class,
+    };
+
+    if (!validateDictionary(script, nil, requiredKeys, keyTypes, outExceptionString))
+        return false;
+
+    if (!validateTarget(script[targetKey], outExceptionString))
+        return false;
+
+    if (script[functionKey] && script[funcKey]) {
+        *outExceptionString = toErrorString(nil, nil, @"Both 'func' and 'function' cannot be passed. Please use 'func'.");
+        return false;
+    }
+
+    if (script[argumentsKey] && script[argsKey]) {
+        *outExceptionString = toErrorString(nil, nil, @"Both 'args' and 'arguments' cannot be passed. Please use 'args'.");
+        return false;
+    }
+
+    bool functionWasPassed = script[functionKey] || script[funcKey];
+    if ((script[filesKey] && functionWasPassed)
+        || (!functionWasPassed && !script[filesKey])) {
```

Comment:
One line would be easier to read.

Reply:
oops, didn't mean to put on two lines, thanks!